launcher/internal/oci/interface.go

```diff
@@ -0,0 +1,71 @@
+// Package oci contains functionalities to interact with OCI image signatures.
+// https://github.com/opencontainers/image-spec/tree/main#readme.
+package oci
+
+import (
+	"go.uber.org/multierr"
+)
+
+// SigningAlgorithm is a specific type for string constants used for sigature signing and verification.
+type SigningAlgorithm string
+
+const (
+	// RSASSAPSS2048SHA256 is RSASSA-PSS 2048 bit key with a SHA256 digest supported for cosign sign.
+	RSASSAPSS2048SHA256 SigningAlgorithm = "RSASSA_PSS_2048_SHA256"
+	// RSASSAPSS3072SHA256 is RSASSA-PSS 3072 bit key with a SHA256 digest supported for cosign sign.
+	RSASSAPSS3072SHA256 SigningAlgorithm = "RSASSA_PSS_3072_SHA256"
+	// RSASSAPSS4096SHA256 is RSASSA-PSS 4096 bit key with a SHA256 digest supported for cosign sign.
+	RSASSAPSS4096SHA256 SigningAlgorithm = "RSASSA_PSS_4096_SHA256"
+	// RSASSAPKCS1V152048SHA256 is RSASSA-PKCS1 v1.5 2048 bit key with a SHA256 digest supported for cosign sign.
+	RSASSAPKCS1V152048SHA256 SigningAlgorithm = "RSASSA_PKCS1V15_2048_SHA256"
+	// RSASSAPKCS1V153072SHA256 is RSASSA-PKCS1 v1.5 3072 bit key with a SHA256 digest supported for cosign sign.
+	RSASSAPKCS1V153072SHA256 SigningAlgorithm = "RSASSA_PKCS1V15_3072_SHA256"
+	// RSASSAPKCS1V154096SHA256 is RSASSA-PKCS1 v1.5 4096 bit key with a SHA256 digest supported for cosign sign.
+	RSASSAPKCS1V154096SHA256 SigningAlgorithm = "RSASSA_PKCS1V15_4096_SHA256"
+	// ECDSAP256SHA256 is ECDSA on the P-256 Curve with a SHA256 digest supported for cosign sign.
+	ECDSAP256SHA256 SigningAlgorithm = "ECDSA_P256_SHA256"
+)
+
+// Signature represents a single OCI image signature.
+type Signature interface {
```

Comment:
Why do we need this interface (like where is this defined externally)?
Are we going to have other implementation other than cosign?

Reply:
There is no other external interface. The intention to define this interface was to align with the API protos we defined for ContainerImageSignature (see https://source.corp.google.com/piper///depot/google3/google/cloud/confidentialcomputing/v1main/service.proto;rcl=551011596;l=315). This interface is more generic and compatible for non-cosign generated signatures if we decide to have implementations in the future.